### ISSUE TITLE
[Bug] RWKV7 channel mixing: three kernels read and write past the tensor end when the element count is not a multiple of the block size

### ISSUE THREAD

**Issue by truong-v** [fla-org/flash-linear-attention#1102]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

`fla/ops/rwkv7/channel_mixing.py` sizes its launch grids with `cdiv(numel, BLOCK_SIZE)` — anticipating a partial last block — but three of its kernels do not mask the tail lanes. Any shape whose element count is not a multiple of the block size reads and, worse, **writes** past the end of the tensors involved. All three sites are reachable from the public `channel_mixing_rwkv7` (the RWKV7 FFN):

1. `rwkv_channel_mixing_pow_and_relu` (forward): no mask on its load or store, `BLOCK_SIZE` hardcoded to 4096. A launch at `numel = 4900` stores 3292 floats past the end of the output.

https://github.com/fla-org/flash-linear-attention/blob/2501ac83fd2c1223f3f0393563f2d58255df9b5d/fla/ops/rwkv7/channel_mixing.py#L88-L94

2. `relu_square_bwd_kernel` (backward): same omission — no element count, no masks — writing past the end of `dk` inside `rwkv_channel_mixing_bwd`.

https://github.com/fla-org/flash-linear-attention/blob/2501ac83fd2c1223f3f0393563f2d58255df9b5d/fla/ops/rwkv7/channel_mixing.py#L180-L191

3. `rwkv_mix_bwd_kenel` (backward): this kernel *does* compute `is_valid` and applies it on three of its four accesses, but the `dx_prev` store is masked by `is_first_step` alone. `is_first_step` is a position predicate, not a bounds predicate: for dead tail lanes `seq_idx` cycles back to 0, so they pass it and store at `batch_idx >= batch_size`, past the end of `dx_prev` — which under `inplace=True` is the caller's own `x_prev` buffer.

https://github.com/fla-org/flash-linear-attention/blob/2501ac83fd2c1223f3f0393563f2d58255df9b5d/fla/ops/rwkv7/channel_mixing.py#L235-L242

The corruption is silent (no launch error) and lands on whatever the caching allocator placed after those tensors. In our runs, the first `backward()` on a fresh autotune cache corrupted gradients that had been computed just before by an unrelated reference path: max abs gradient error 6e+01 against the same computation with masks (8e-06). It goes unnoticed at common RWKV7 configurations because `n_embd` and `dim_ffn` are typically powers of two and every element count comes out block-aligned — e.g. `test_channel_mixing_gradients` uses (B, T, n_embd, dim_ffn) = (2, 1024, 1024, 4096), all numels multiples of 8192.

### Steps to Reproduce the Bug

Sentinel-filled guard bands around the tensors each site writes through (element counts chosen so no block size in [128, 8192] divides them):

```python
import torch
import fla.ops.rwkv7.channel_mixing as cm

torch.manual_seed(42)
SENT = 777.0
B, T, n_embd, dim_ffn = 1, 7, 200, 700
N = B * T * dim_ffn                    # 4900
W = (N + 4095) // 4096 * 4096          # 8192, the window the 4096 grid covers

# site 1: forward relu-square, in place on a view of a sentinel-filled buffer
buf = torch.full((2 * W,), SENT, device='cuda')
cm.rwkv_relu_and_square_fwd(buf[:N].view(B, T, dim_ffn), inplace=True)
print('site 1  pow_and_relu     cells written past the end:', (buf[N:] != SENT).sum().item())

# site 2: backward relu-square, launched exactly as rwkv_channel_mixing_bwd launches it
buf = torch.full((2 * W,), SENT, device='cuda')
dk, k1_K = buf[:N], buf[W:W + N]
cm.relu_square_bwd_kernel[((N + 4095) // 4096,)](dk, k1_K, BLOCK_SIZE=4096)
print('site 2  relu_square_bwd  cells written past dk:', (buf[N:W] != SENT).sum().item())

# site 3: full backward with inplace=True, so dx_prev is the caller's x_prev
big = torch.full((1 << 20,), SENT, device='cuda')
x = torch.randn(B, T, n_embd, device='cuda')
x_prev = big[:B * n_embd].view(B, n_embd)
x_k = torch.randn(1, 1, n_embd, device='cuda')
Kw = torch.randn(n_embd, dim_ffn, device='cuda')
Vw = torch.randn(dim_ffn, n_embd, device='cuda')
k1 = cm.rwkv_mix_fwd(x, x_prev, x_k)
k1_K = k1 @ Kw
k = cm.rwkv_relu_and_square_fwd(k1_K, inplace=False)
go = torch.randn(B, T, n_embd, device='cuda')
cm.rwkv_channel_mixing_bwd(go, x, x_prev, x_k, Kw, Vw, k1, k1_K, k, inplace=True)
print('site 3  mix_bwd dx_prev  cells written past dx_prev:', (big[B * n_embd:] != SENT).sum().item())
```

```
site 1  pow_and_relu     cells written past the end: 3292
site 2  relu_square_bwd  cells written past dk: 3292
site 3  mix_bwd dx_prev  cells written past dx_prev: 8
```

Driving the public `channel_mixing_rwkv7` forward+backward at the same shape under `compute-sanitizer --tool memcheck` with `PYTORCH_NO_CUDA_MEMORY_CACHING=1` attributes every invalid access to the three sites:

```
========= Invalid __global__ read of size 16 bytes
=========     at rwkv_channel_mixing_pow_and_relu+0x110 in fla/ops/rwkv7/channel_mixing.py:91
=========     and is 1 bytes after the nearest allocation at 0x7fa6983c5e00 of size 19,600 bytes
...
error counts by source line, one public forward+backward:
  channel_mixing.py:91    142 invalid reads    (pow_and_relu load)
  channel_mixing.py:94     96 invalid writes   (pow_and_relu store)
  channel_mixing.py:184     23 invalid reads   (relu_square_bwd load)
  channel_mixing.py:185     96 invalid reads   (relu_square_bwd load)
  channel_mixing.py:240  3,248 invalid writes  (mix_bwd dx_prev store)
  ERROR SUMMARY: 3606 errors        -> 0 device errors with the masks in place
```

### Expected Behavior

No access outside the tensors: tail lanes masked, as this file's own `rwkv_seq_mix_kernel` already does on every load and store. PR with the fix and a regression test: #PR_NUMBER.

### Environment Information

1. Torch: 2.13.0+cu129
2. Triton: 3.7.1
3. Commit tested: 2501ac83 (current main), Python 3.12, NVIDIA L40S, driver CUDA 12.4
